sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/TestRegexScannerContext.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.secrets.utils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.sonar.api.batch.fs.InputComponent;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.check.Rule;
+import org.sonar.java.regex.RegexCheck;
+import org.sonar.java.regex.RegexScannerContext;
+import org.sonar.plugins.java.api.JavaCheck;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.JavaVersion;
+import org.sonar.plugins.java.api.caching.CacheContext;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
+import org.sonarsource.analyzer.commons.regex.ast.NonCapturingGroupTree;
+import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement;
+
+public class TestRegexScannerContext implements JavaFileScannerContext, RegexScannerContext {
+
+  private final List<Issue> issues = new ArrayList<>();
+  private SecretsRegexTest.PatternLocation patternLocation;
+
+  public void verify() {
+    if (!issues.isEmpty()) {
+      StringBuilder message = new StringBuilder("Found following issues in Regexes:\n");
+      for (Issue issue : issues) {
+        var msg = String.format("%s, id: %s, location: %s, the regex: `%s`, violating rule: %s: %s",
```

Comment:
IMO, this could benefit from the following:
```suggestion
        var msg = String.format("%s, id: %s, location: %s, the regex:\n`%s`\n violating rule: %s: %s",
```